### ISSUE TITLE
Infinite loop when accessing account management console

### ISSUE THREAD

**Issue by Nicolazic** [keycloak/keycloak#33109]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

account/ui

### Describe the bug

Hello,

We set up Keycloak with docker-compose and Apache proxy.
The admin console is accessible and working fine.
We are able to connect to Nextcloud or Wordpress thanks to Keyloaker.
But when we try to go to https://\<url\>/realms/master/account, the page is infinitly loading (see screenshot).

![image](https://github.com/user-attachments/assets/b8edef2c-5914-42f3-be4b-e8fa7d7be931)

When i open the console, it says this : 

![image](https://github.com/user-attachments/assets/776f9885-2980-4171-a12b-da7bec6ec217)

--------------------------------

Docker compose config :

keycloak:
    image: quay.io/keycloak/keycloak:latest
    container_name: keycloak
    env_file: .env
    environment:
      - KC_DB=postgres
      - KC_DB_URL=jdbc:postgresql://keycloak_db:5432/$KC_DB_NAME
      - KC_DB_USERNAME=$KC_DB_USER
      - KC_DB_PASSWORD=$KC_DB_PASS
      - KC_HOSTNAME=https://\<url\>
      - KC_HOSTNAME_PORT=8080
      - KC_HOSTNAME_STRICT=false
      - KC_HOSTNAME_STRICT_HTTPS=false
      - KC_LOG_LEVEL=info
      - KC_METRICS_ENABLED=true
      - KC_HEALTH_ENABLED=true
      - KEYCLOAK_ADMIN=$ADMIN_NAME
      - KEYCLOAK_ADMIN_PASSWORD=$ADMIN_PASS
    command: start-dev
    depends_on:
      - keycloak_db
    ports:
      - 19830:8080
    networks:
      - xxx

--------------------------------

Apache config :  

\<VirtualHost *\:80\>
    ServerName \<url\>
    Redirect permanent / https://\<url\>
\<\/VirtualHost\>

\<VirtualHost *:443\>
    ServerName \<url\>

    Header always set Strict-Transport-Security "max-age=63072000; includeSubdomains; preload"

    SSLEngine ON
    SSLProtocol all -SSLv2 -SSLv3
    SSLCipherSuite EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH
    SSLOptions +FakeBasicAuth +ExportCertData +StrictRequire
    SSLCertificateFile /etc/letsencrypt/live/<url>/cert.pem
    SSLCertificateKeyFile /etc/letsencrypt/live/<url>/privkey.pem
    SSLCertificateChainFile /etc/letsencrypt/live/<url>/chain.pem

    ProxyRequests Off
    ProxyPreserveHost On
    ProxyPass / http://127.0.0.1:19830/ connectiontimeout=60 timeout=60 retry=1 Keepalive=On
    ProxyPassReverse / http://127.0.0.1:19830/

    # Logs
    CustomLog "/var/log/apache2/wordpress/access" common
    ErrorLog "/var/log/apache2/wordpress/error"
\<\/VirtualHost\>

### Version

25.0.5

### Regression

- [ ] The issue is a regression

### Expected behavior

Access the Account Management Console.

### Actual behavior

Infinite loop when accessing the Account Management Console.

### How to Reproduce?

Go to https://\<url\>/realms/master/account

**Comment by Nicolazic** [comment]
Ok I found the solution here : https://keycloak.discourse.group/t/status-code-403-forbidden/10854
Very weird issue though...

**Comment by shawkins** [comment]
Setting the web origins to * is probably not desirable. It would be better to determine why the inferred origin does not match what is allowed by default.  

Similar to #31946 but what is different here is that proxy-headers is not set, so it's clearly not a problem with the header. These issues make the case for adding more logging around that logic, so that it's easier to understand what is happening and how to address it.

Note also with hostname v2 in KC 25+ KC_HOSTNAME_PORT and KC_HOSTNAME_STRICT are effectively ignored when using a full url for the hostname, and KC_HOSTNAME_STRICT_HTTPS is no longer used.

**Comment by Nicolazic** [comment]
Ok so I will clean the KC_HOSTNAME_* from my config and anly keep KC_HOSTNAME, I didn't know about this changement.
The origin parameter is the KC_HOSTNAME value (but url encoded).

I'm noticing another wrong behavior. When I'm logged-in with a normal user (not admin), and try to access "KC_HOSTNAME /admin/master/console", I also get an infinite loop waiting animation and the error in networks : 

![image](https://github.com/user-attachments/assets/90f1ae51-54b2-4755-8453-2b37512a7aa5)

**Comment by shawkins** [comment]
> The origin parameter is the KC_HOSTNAME value (but url encoded).

Thanks for the confirmation. At first glance that should work. Can you please post the output from using --hostname-debug as well?

> I'm noticing another wrong behavior. When I'm logged-in with a normal user (not admin), and try to access "KC_HOSTNAME /admin/master/console", I also get an infinite loop waiting animation and the error in networks

This is similar to #32847 but manifesting in a different way.

**Comment by shawkins** [comment]
@thomas-zic this was closed to the the pr associated with additional logging on 403 from init. If you provide additional information confirming there is a deeper issue here, then this can be reopened, or a new issue created.

**Comment by thomas-zic** [comment]
I tried with '--hostname-debug true' and put back an empty web origin in the account console client, but it returns the exact same two networks screens I shared previously.

**Comment by shawkins** [comment]
@thomas-zic We're looking for the output from ```http://mykeycloak:8080/realms/master/hostname-debug``` see https://www.keycloak.org/server/hostname

**Comment by thomas-zic** [comment]
![image](https://github.com/user-attachments/assets/e8214c05-7230-4c1d-b07b-28389f0ff6a0)

**Comment by shawkins** [comment]
@thomas-zic presumably this request was via the proxy correct? It shows the requst, frontend, admin, hostname, and Host header as all starting with the same https hostname value?

**Comment by thomas-zic** [comment]
Yes, all the values grayed out in purple are the hostname value.

**Comment by thomas-zic** [comment]
That's working !
I put KC_PROXY_HEADERS=xforwarded in my Docker Compose and I delete the "*" in web origin in the account console client.
But is it a normal behavior that my non-admin user always see an infinite loop animation when accessing /admin/master/console ? I think it would be better to redirect him elsewhere accessible.

**Comment by shawkins** [comment]
> But is it a normal behavior that my non-admin user always see an infinite loop animation when accessing /admin/master/console ? I think it would be better to redirect him elsewhere accessible.

Ideally this should be covered by #32847
